src/tribler-common/tribler_common/reported_error.py

```diff
@@ -0,0 +1,14 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class ReportedError:
```

Comment:
**praise**: With this class, the code looks much clearer now!

Reply:
🙏